lib/LDA_CTR.py

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env python
+"""
+A module that contains the content-based recommender LDARecommender that uses
+LDA.
+"""
+from lib.content_based import ContentBased
+from overrides import overrides
+from sklearn.decomposition import LatentDirichletAllocation
+import numpy
+from numpy.linalg import solve
+
+
+class LDACTRRecommender(ContentBased):
+    """
+    LDA Recommender, a content based recommender that uses LDA.
+    """
+    def __init__(self, initializer, evaluator, hyperparameters, options,
+                 verbose=False, load_matrices=True, dump_matrices=True):
+        """
+        Constructor of Latent Dirichilet allocation's processor.
+
+        :param ModelInitializer initializer: A model initializer.
+        :param Evaluator evaluator: An evaluator of recommender and holder of input.
+        :param dict hyperparameters: A dictionary of the hyperparameters.
+        :param dict options: A dictionary of the run options.
+        :param boolean verbose: A flag for printing while computing.
+        :param boolean load_matrices: A flag for reinitializing the matrices.
+        :param boolean dump_matrices: A flag for saving the matrices.
+        """
+        super(LDACTRRecommender, self).__init__(initializer, evaluator, hyperparameters, options,
+                                             verbose, load_matrices, dump_matrices)
+
+    def train_k_fold(self):
+        all_errors = []
+        for current_k in range(self.k_folds):
+            self.set_data(*self.evaluator.get_fold(current_k, self.fold_test_indices))
+            self.hyperparameters['fold'] = current_k
+            self.train_one_fold(False)
+            all_errors.append(self.get_evaluation_report())
```

Comment:
all_errors.append(self.train_one_fold())
should be the same like the last 2 lines